python/cudf/cudf/core/dataframe.py

```diff
@@ -2876,7 +2876,7 @@ def partition_by_hash(self, columns, nparts, keep_index=True):
         # Slice into partition
         return [outdf[s:e] for s, e in zip(offsets, offsets[1:] + [None])]
 
-    def replace(self, to_replace, replacement):
+    def replace(self, to_replace, replacement, inplace=False):
```

Comment:
Can this function just be moved down to the `Frame` level? Looks like we should be able to basically just loop over the columns given in `to_replace` / `replacement` regardless. Maybe keep a much shorter implementation in Series / DataFrame / Index / MultiIndex that just validates the shape of the inputs.

Reply:
Pushed down the core logic to `Frame` level. In addition to this change, made additional changes to fix: #3787 and also support a no parameter call to `replace`, like:
`cudf.Series([1,2,3]).replace()`